wandb/apis/importers/wandb.py

```diff
@@ -0,0 +1,1879 @@
+
+                new_art = _clone_art(art)
+                group = [new_art]
+                run = WandbRun(wandb_run)
+
+            internal.send_artifacts_with_send_manager(
+                group,
+                run,
+                overrides=namespace.send_manager_overrides,
+                settings_override=settings_override,
+                config=send_manager_config,
+            )
+            logger.info(
+                f"Uploaded partial artifact sequence, {i}/{len(groups_of_artifacts)}"
+            )
+        logger.info(f"Finished uploading {seq=}")
+
+        # query it back and remove placeholders
+        self._remove_placeholders(art)
+
+    def _remove_placeholders(self, art: Artifact) -> None:
+        name, _ = _get_art_name_ver(art)
+        try:
+            dst_versions = list(self.dst_api.artifacts(art.type, name))
+        except wandb.CommError:
+            # the artifact did not upload for some reason
+            logger.warn(f"Art doesn't exist in dst (already deleted?), {art=}")
+            return
+        except TypeError:
+            logger.error("Problem getting dst versions (try again later)")
+            return
+
+        for version in dst_versions:
+            if version.description != ART_SEQUENCE_DUMMY_PLACEHOLDER:
+                continue
+            if version.type in ("wandb-history", "job"):
+                continue
+
+            try:
+                version.delete(delete_aliases=True)
+            except wandb.CommError as e:
+                if "cannot delete system managed artifact" in str(e):
+                    logger.warn("Cannot delete system managed artifact")
+                else:
+                    raise e
+
+    def _compare_artifact_dirs(self, src_dir, dst_dir):
+        def compare(src_dir, dst_dir):
+            comparison = filecmp.dircmp(src_dir, dst_dir)
+            differences = {
+                "left_only": comparison.left_only,  # Items only in dir1
+                "right_only": comparison.right_only,  # Items only in dir2
+                "diff_files": comparison.diff_files,  # Different files
+                "subdir_differences": {},  # Differences in subdirectories
+            }
+
+            # Recursively find differences in subdirectories
+            for subdir in comparison.subdirs:
+                subdir_differences = compare(
+                    os.path.join(src_dir, subdir), os.path.join(dst_dir, subdir)
+                )
+                # If there are differences, add them to the result
+                if any(subdir_differences.values()):
+                    differences["subdir_differences"][subdir] = subdir_differences
+
+            if all(not diff for diff in differences.values()):
+                return None
+
+            return differences
+
+        return compare(src_dir, dst_dir)
+
+    def _compare_artifact_manifests(self, src_art: Artifact, dst_art: Artifact):
+        problems = []
```

Comment:
regarding `problems`: I really like the concept of telling the user what exactly went wrong with the import, but I would add a test for that, seems really valuable given the complexity of this code and all the interactions with the backend/fe.